
fix speaker-test -c 6 resulting silence on FC/RR/LFE

buffer is u8, and expected data u32, so the &buffer[x] offset seem strange and wrong info is sent in audio info frames

fix developed by kwiboo

diff --speed-large-files --no-dereference --minimal -Naur linux-6.18-rc7/drivers/gpu/drm/bridge/synopsys/dw-hdmi-qp.c linux-6.18-rc7/drivers/gpu/drm/bridge/synopsys/dw-hdmi-qp.c
--- linux-6.18-rc7/drivers/gpu/drm/bridge/synopsys/dw-hdmi-qp.c	2025-11-28 12:52:23.940321824 +0100
+++ linux-6.18-rc7/drivers/gpu/drm/bridge/synopsys/dw-hdmi-qp.c	2025-11-28 12:51:57.120321101 +0100
@@ -920,7 +920,7 @@
 
 	regmap_bulk_write(hdmi->regm, PKT_AUDI_CONTENTS0, &header_bytes, 1);
 	regmap_bulk_write(hdmi->regm, PKT_AUDI_CONTENTS1, &buffer[3], 1);
-	regmap_bulk_write(hdmi->regm, PKT_AUDI_CONTENTS2, &buffer[4], 1);
+	regmap_bulk_write(hdmi->regm, PKT_AUDI_CONTENTS2, &buffer[7], 1);
 
 	/* Enable ACR, AUDI, AMD */
 	dw_hdmi_qp_mod(hdmi,
